api.ts

```diff
@@ -1847,31 +1879,46 @@ export type PropertyOptionsMetadataFormula<SchemaT extends Schema> = ObjectPackF
   execute(params: ParamValues<[]>, context: PropertyOptionsExecutionContext): Promise<object> | object;
 };
 
-export type MetadataFormulaMetadata = Omit<MetadataFormula, 'execute'>;
+export type MetadataFormulaMetadata = Omit<MetadataFormula, 'execute' | 'validateParameters'>;
 
 /**
  * @hidden
  */
-export declare type GenericMetadataFormulaMetadata = Omit<GenericMetadataFormula, 'execute'>;
+export declare type GenericMetadataFormulaMetadata = Omit<GenericMetadataFormula, 'execute' | 'validateParameters'>;
+
+/**
+ * Historically, metadata formulas could return a variety of types.
+ * This type is used to represent the legacy return types.
+ * Metadata formulas now are generic and can return any value.
```

Comment:
Nit, maybe "Metadata formulas now work with generics and in the future will be updated to declare a more precise return type for each formula."